app/decorators/order_decorator.rb

```diff
@@ -15,4 +15,26 @@ def state_badge
       helpers.tag.span('failed', class: 'order-state badge badge-danger')
     end
   end
+
+  def events_buttons
+    permitted_events = Orders::StateService.new(object).permitted_events
+    %w[complete fail].each do |event|
+      h.concat method("#{event}_button").call if permitted_events.include?(event.to_sym)
+    end
+    h.concat not_available if permitted_events.empty?
```

Comment:
it should be returned before iteration -> if permitted_events are empty, iteration also will be empty.

Reply:
fixed